pkg/processor/eventprocessor/pool.go

```diff
@@ -18,32 +18,27 @@ package eventprocessor
 
 import (
 	"context"
-	"fmt"
 	"sync/atomic"
 	"time"
 
 	"github.com/nuclio/nuclio/pkg/common"
-	"github.com/nuclio/nuclio/pkg/errgroup"
 	"github.com/nuclio/nuclio/pkg/processor/statistics"
 
 	"github.com/nuclio/errors"
 	"github.com/nuclio/logger"
 )
 
 type syncPoolAllocator struct {
-	logger      logger.Logger
+	*abstractPoolAllocator
 	objectsChan chan EventProcessor
-	objects     []EventProcessor
 	statistics  safeAllocatorStatistics
-
-	isTerminated atomic.Bool
 }
 
 func NewSyncPoolAllocator(parentLogger logger.Logger, objects []EventProcessor) (Allocator, error) {
+	abstractPoolAllocatorInstance := newAbstractPoolAllocator(parentLogger.GetChild("sync_pool_allocator"))
```

Comment:
This is not super aligned in all of the codebase, but lets keep logger names either camelCase or with `-`.
```suggestion
	abstractPoolAllocatorInstance := newAbstractPoolAllocator(parentLogger.GetChild("sync-pool-allocator"))
```